dvc/repo/experiments/save.py

```diff
@@ -0,0 +1,32 @@
+import logging
+import os
+from typing import TYPE_CHECKING, Optional
+
+from funcy import first
+
+if TYPE_CHECKING:
+    from dvc.repo import Repo
+
+
+logger = logging.getLogger(__name__)
+
+
+def save(
+    repo: "Repo",
+    name: Optional[str] = None,
+    force: bool = False,
+) -> Optional[str]:
+    """Save the current workspace status as an experiment.
+
+    Returns the saved experiment's SHAs.
+    """
+    queue = repo.experiments.workspace_queue
+    logger.debug("Saving workspace in %s", os.getcwd())
+
+    entry = repo.experiments.new(queue=queue, name=name, force=force)
+    executor = queue.init_executor(repo.experiments, entry)
+    save_result = executor.save(executor.info, force=force)
+    result = queue.collect_executor(repo.experiments, executor, save_result)
```

Comment:
Feels a little odd to involve stuff like executor and queue when we are not really going to execute or queue anything. 

I get that is because `_stash_exp` is currently part of the `queue` implementation and it might be out of the scope of the P.R. to refactor, but wanted to discuss it.

Reply:
The workspace queue is already a dummy queue that only ever contains one item.

I think this actually does work well since using the full executor init/collect workflow this way sets up required git refs for generating exp commits and refs (and cleans up the git workspace when we are done). Doing it this way also ensures that the workspace state is restored in the event that something in DVC breaks during the actual save commit operation(s).